rueidisotel/metrics.go

```diff
@@ -0,0 +1,175 @@
+package rueidisotel
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"net"
+	"sync"
+	"time"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/metric"
+
+	"github.com/redis/rueidis"
+)
+
+const metricNamespace = "github.com/redis/rueidis"
+
+var DefaultDialLatencyHistogramDefaultBucketBoundaries = []float64{
+	.005, .01, .025, .05, .1, .25, .5, 1, 2.5, 5, 10,
+}
+
+var ErrDialFnRequired = errors.New("rueidisotel: clientOption.DialFn is required")
```

Comment:
Could we set a default DialFn instead of throwing this error?

Reply:
OK. I Change it. [e026def](https://github.com/redis/rueidis/pull/421/commits/e026def866af3c98a2fbdcc4fdc691b27e0d8da7)